src/index.ts

```diff
@@ -50,17 +52,145 @@ export default class WorkspaceClient {
     }
 
     private static createAxiosInstance(config: IRestAPIConfig): AxiosInstance {
-        if (config.ssCrtPath && this.isItNode() && fs.existsSync(config.ssCrtPath)) {
-            const agent = new https.Agent({
-                ca: fs.readFileSync(config.ssCrtPath)
-            });
-            return axios.create({httpsAgent: agent});
+        if (! this.isItNode()) {
+            return axios;
+        }
+
+        let certificateAuthority : Buffer | undefined;
+        if (config.ssCrtPath && fs.existsSync(config.ssCrtPath)) {
+            certificateAuthority = fs.readFileSync(config.ssCrtPath);
         }
 
+        const proxyUrl = process.env.http_proxy;
+        if (proxyUrl && proxyUrl !== '' && config.baseUrl) {
+            let mainProxyOptions: tunnel.ProxyOptions;
+            var parsedProxyUrl = url.parse(proxyUrl);
+            let port = 3128;
+            if (parsedProxyUrl.port && parsedProxyUrl.port !== '') {
+                port = Number(parsedProxyUrl.port);
+            }
+            mainProxyOptions = {
+                host: parsedProxyUrl.hostname!,
+                port: port
+            };
+
+            if (parsedProxyUrl.auth && parsedProxyUrl.auth !== '') {
+                mainProxyOptions.proxyAuth = parsedProxyUrl.auth;
+            }
+
+            const noProxyEnv = process.env.no_proxy || process.env.NO_PROXY;
+            let noProxy: string[] = [];
+            if (noProxyEnv) {
+                noProxy = noProxyEnv.split(',').map(function trim(s: string) {
+                    return s.trim();
+                });
+            }
+
+            var parsedBaseUrl = url.parse(config.baseUrl);
```

Comment:
Do we really need the function scoped variable here?

Reply:
@vinokurig Could you have a look to this one ? I think you would have a clearer thought than me on this.